src/NodeGenerators/fillFacetedVolume.cc

```diff
@@ -202,6 +242,24 @@ fillFacetedVolume10(const Dim<3>::FacetedVolume& outerBoundary,
       result.insert(result.end(), result_thread.begin(), result_thread.end());
     }
   }
+
+  // If we didn't find anything, fall back to sampling on the surface.
+  if (allReduce(result.size(), MPI_SUM, Communicator::communicator()) == 0U) {
+    if (Process::getRank() == 0) {
+      cerr << "Falling back to surface points..." << endl;
+      const auto nexpect = std::max(1, std::min(int(verts.size()), int(outerBoundary.volume()/(dx*dx*dx) + 0.5)));
+      std::vector<unsigned> iresult(verts.size());
+      for (auto i = 0U; i < verts.size(); ++i) iresult[i] = i;
+      while (iresult.size() > nexpect) {
```

Comment:
A sign warning is thrown between `iresult.size()` and `nexpect` on 253. `nexpect` is evaluated as an integer from `std::max`, since it will never be negative could we make it unsigned?

Reply:
Fixed in the latest commit.